src/Extensions/Diagnostics.HealthChecks/Internal/HttpEndpointHealthCheck.cs

```diff
@@ -65,5 +66,23 @@ protected override async Task<HealthCheckResult> CheckHealthInternalAsync(Health
 
 			return result;
 		}
+
+		private static HttpRequestMessage CloneRequestMessage(HttpRequestMessage message)
+		{
+			HttpRequestMessage clone = new()
+			{
+				Method = message.Method,
+				RequestUri = message.RequestUri,
+				Content = message.Content,
+				Version = message.Version
+			};
+
+			foreach (KeyValuePair<string, IEnumerable<string>> header in message.Headers)
+			{
+				clone.Headers.TryAddWithoutValidation(header.Key, header.Value);
+			}
+
+			return clone;

```

Comment:
Should we also copy [Options](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestoptions?view=net-5.0) and [VersionPolicy ](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.versionpolicy?view=net-5.0#System_Net_Http_HttpRequestMessage_VersionPolicy)or we are not planning to use them?